jscomp/runtime/js_typed_array.ml

```diff
@@ -373,3 +373,77 @@ module Float64Array = struct
 end
 module Float64_array = Float64Array
 [@ocaml.deprecated "use `Float64Array` instead"]
+
+module DataView = struct
+  (** The DataView view provides a low-level interface for reading and writing
+      multiple number types in an ArrayBuffer irrespective of the platform's endianness.
+
+    @see <https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/DataView> MDN
+  *)
+
+  type t
+
+  external make : array_buffer -> t = "DataView" [@@bs.new]
+  external fromBuffer : array_buffer -> t = "DataView" [@@bs.new]
+  external fromBufferOffset : array_buffer -> int -> t = "DataView" [@@bs.new]
+  external fromBufferRange : array_buffer -> offset:int -> length:int -> t = "DataView" [@@bs.new]
+
+  external buffer : t -> array_buffer = "" [@@bs.get]
+  external byteLength : t -> int = "" [@@bs.get]
+  external byteOffset : t -> int = "" [@@bs.get]
+
+  external getInt8 : t -> int -> int = "" [@@bs.send]
+  external getUint8 : t -> int -> int = "" [@@bs.send]
+
+  external getInt16 : t -> int -> littleEndian:Js.boolean -> int = "" [@@bs.send]
```

Comment:
I tried to make it explicit as much as can =) But probably in real code big endian is the most wanted and good to have it in short form, you are right.